daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/DAFFunctions.scala

```diff
@@ -72,3 +78,63 @@ case object DAFError extends RecipeOp {
     }
   }
 }
+
+case class DAFLookAhead(recipes: List[CompiledDPath])
+  extends FNTwoArgs(recipes) {
+
+  def computeValue(arg1: AnyRef, arg2: AnyRef, dstate: DState): AnyRef = {
+    val offset = arg1.asInstanceOf[JLong]
+    val lBitSize = arg2.asInstanceOf[JLong]
+
+    /*
+     * Since daf:lookAhead is defined to take unsigned arguements, the DPath interperater
+     * will error out on the cast if a negative arguement is supplied, so we do not need to SDE here.
+     */
+
+    Assert.invariant(offset >= 0)
+    Assert.invariant(lBitSize >= 0)
+
+    val totalLookahead = offset + lBitSize
+    if (totalLookahead > 512) {
```

Comment:
This should be a constant variable, or maybe even a tunable? I don't see anything code that actually has a limit of 512, so it seems this is just to prevent people from accidentally doing the wrong thing or abusing this function. A tunable probably make sense in that case, so if someone reeeeeally wants to abuse this they can.

Reply:
My main motivation for this limit was to be nice to future implementations where infinite lookahead might be less trivial. An added benifit is that it prevents people from shooting themselves in the foot with a long lookahead on streamed input.

Adding the tunable maxDAFLookaheadBits